public/app/plugins/panel/canvas/editor/element/ButtonStyleEditor.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useCallback } from 'react';
+
+import { SelectableValue, StandardEditorProps } from '@grafana/data';
+import { ButtonVariant, InlineField, InlineFieldRow, RadioButtonGroup } from '@grafana/ui';
+import { defaultStyleConfig } from 'app/features/canvas/elements/button';
+
+export interface ButtonStyleConfig {
+  variant: ButtonVariant;
+}
+
+type Props = StandardEditorProps<ButtonStyleConfig>;
+
+const variantOptions: SelectableValue[] = [
+  { label: 'primary', value: 'primary' },
+  { label: 'secondary', value: 'secondary' },
+  { label: 'success', value: 'success' },
+  { label: 'destructive', value: 'destructive' },
+];
+
+export const ButtonStyleEditor = ({ value, onChange }: Props) => {
+  if (!value) {
+    value = defaultStyleConfig;
+  }
+
+  const onVariantChange = useCallback(
+    (variant: ButtonVariant) => {
+      onChange({
+        ...value,
+        variant,
+      });
+    },
+    [onChange, value]
+  );
+
+  return (
+    <>
+      <InlineFieldRow>
+        <InlineField label="Method" grow={true}>
+          <RadioButtonGroup value={value?.variant} options={variantOptions} onChange={onVariantChange} fullWidth />
```

Comment:
the radio button group doesn't scale super great on smaller screens, especially for more than 2-3 options 🤔 would maybe a dropdown make more sense for this?


https://github.com/grafana/grafana/assets/22381771/7254e629-91cc-48d2-9568-e057158cd4b2